ee/tabby-db/src/smtp_creds.rs

```diff
@@ -0,0 +1,53 @@
+use anyhow::Result;
+use rusqlite::OptionalExtension;
+
+use crate::DbConn;
+
+pub struct SMTPInfo {
+    pub email: String,
+    pub password: String,
+    pub mailserver_url: String,
+}
+
+impl SMTPInfo {
+    fn new(email: String, password: String, mailserver_url: String) -> Self {
+        Self {
+            email,
+            password,
+            mailserver_url,
+        }
+    }
+}
+
+impl DbConn {
```

Comment:
Please refer https://github.com/TabbyML/tabby/blob/main/ee/tabby-db/src/github_oauth_credential.rs on how to properly implement the read / update logic for a single row table. (hint: use a static row id)